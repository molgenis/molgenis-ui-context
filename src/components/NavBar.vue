<template>
  <nav class="navbar navbar-expand-md navbar-light bg-light">

    <a v-if="molgenisMenu.navBarLogo" class="navbar-brand"
       :href="'/menu/main/'+ molgenisMenu.menu.items[0].href">
      <img :src="molgenisMenu.navBarLogo" height="20">
    </a>
    <a v-else class="navbar-brand" href="#"></a>

    <button class="navbar-toggler collapsed" type="button" data-toggle="collapse" data-target="#navbar-content"
            aria-controls="navbar-content" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div class="collapse navbar-collapse" id="navbar-content">
      <ul class="navbar-nav mr-auto">

        <template v-for="item in molgenisMenu.menu.items">
          <li v-if="item.type === 'PLUGIN' && item.id !== 'home'" :key="item.id"
              :class="['nav-item', {'active': isSelectedPlugin(item.id)}]">

            <a class="nav-link" :href="'/menu/main/'+item.href">
              {{ item.label }}
            </a>
          </li>

          <li v-else-if="item.id !== 'home'" class="nav-item dropdown" :key="item.id">
            <a class="nav-link dropdown-toggle" :id="item.id" data-toggle="dropdown" aria-haspopup="true"
               aria-expanded="false">
              {{ item.label }}
            </a>
            <drop-down-items :parent="item" :items="item.items"/>
          </li>
        </template>
      </ul>

      <ul class="navbar-nav justify-content">
        <li v-if="molgenisMenu.authenticated && languages.length > 1" class="nav-item">
          <form id="language-form" class="navbar-form">
            <select class="nav-link" v-model="selectedLanguage" @change="handleLanguageSelect">
              <option
                v-for="language in languages"
                :key="language.id"
                :value="language.id"
                :selected="language.id === selectedLanguage"
              >
                {{ language.label }}
              </option>
            </select>
          </form>
        </li>
      </ul>

      <ul class="navbar-nav justify-content-end">
        <li class="nav-item">
          <a class="nav-link" :href="molgenisMenu.helpLink.href" target="_blank">
            {{ molgenisMenu.helpLink.label }}
          </a>
        </li>

        <li class="nav-item">
          <form id="logout-form" class="form-inline" method="post" action="/logout">
            <button v-if="molgenisMenu.authenticated" id="signout-button" type="button" class="btn btn-outline-secondary"
                    @click="logout">
              Sign out
            </button>

            <a v-else class='btn btn-outline-primary' href="/login">Sign in</a>
          </form>
        </li>
      </ul>
    </div>
  </nav>
</template>

<script>
import Vue from 'vue'
import { MolgenisMenu } from '../types'
import DropDownItems from './DropDownItems'
import api from '@molgenis/molgenis-api-client'

export default Vue.extend({
  name: 'NavBar',
  props: {
    molgenisMenu: MolgenisMenu
  },
  components: {
    DropDownItems
  },
  data () {
    return {
      selectedLanguage: null,
      languages: [],
      helperStyle: this.molgenisMenu.topLogo ? {
        display: 'inline-block',
        height: '100%',
        verticalAlign: 'middle'
      } : undefined
    }
  },
  computed: {
    topLogoStyle () {
      return this.molgenisMenu.topLogo ? {
        maxHeight: this.molgenisMenu.topLogoMaxHeight + 'px',
        verticalAlign: 'middle'
      } : undefined
    },
    topLogoVueBannerStyle () {
      return this.molgenisMenu.topLogo ? {
        height: this.molgenisMenu.topLogoMaxHeight + 'px'
      } : undefined
    }
  },
  methods: {
    isSelectedPlugin (plugin) {
      return plugin === this.selectedPlugin
    },
    logout () {
      if (this.logoutFunction) {
        this.logoutFunction()
      }
      document.getElementById('logout-form').submit()
    },
    handleLanguageSelect () {
      api.post('/plugin/useraccount/language/update?languageCode=' + this.selectedLanguage).then(() => {
        location.reload(true)
      })
    }
  },
  mounted () {
    if (this.molgenisMenu.authenticated) {
      api.get('/api/v2/sys_Language?q=active==true').then(response => {
        this.languages = response.items.map(item => {
          const language = { id: item.code, label: item.name }
          if (item.code === response.meta.languageCode) {
            this.selectedLanguage = language.id
          }
          return language
        })
      }, error => {
        console.error(error)
      })
    }
  }
})
</script>
