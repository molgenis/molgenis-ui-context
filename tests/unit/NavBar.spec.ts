import { shallowMount } from '@vue/test-utils'
import NavBar from '@/components/NavBar.vue'
import molgenisMenu from '@/exampleMenu'
jest.mock('@molgenis/molgenis-api-client')

describe('NavBar.vue', () => {
  it('should match the snapshot', () => {
    const wrapper = shallowMount(NavBar, {
      propsData: { molgenisMenu }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
