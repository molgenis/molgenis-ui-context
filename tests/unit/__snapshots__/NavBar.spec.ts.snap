// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar.vue should match the snapshot 1`] = `
<nav class="navbar navbar-expand-md navbar-light bg-light"><a href="/menu/main/home" class="navbar-brand"><img src="/img/logo_molgenis_small.png" height="20"></a> <button type="button" data-toggle="collapse" data-target="#navbar-content" aria-controls="navbar-content" aria-expanded="false" aria-label="Toggle navigation" class="navbar-toggler collapsed"><span class="navbar-toggler-icon"></span></button>
  <div id="navbar-content" class="collapse navbar-collapse">
    <ul class="navbar-nav mr-auto">
      <!---->
      <li class="nav-item dropdown"><a id="importdata" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" class="nav-link dropdown-toggle">
          Import data
        </a>
        <drop-down-items-stub parent="[object Object]" items="[object Object],[object Object]" depth="0"></drop-down-items-stub>
      </li>
      <li class="nav-item"><a href="/menu/main/navigator" class="nav-link">
          Navigator
        </a></li>
      <li class="nav-item"><a href="/menu/main/dataexplorer" class="nav-link">
          Data Explorer
        </a></li>
      <li class="nav-item dropdown"><a id="dataintegration" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" class="nav-link dropdown-toggle">
          Data Integration
        </a>
        <drop-down-items-stub parent="[object Object]" items="[object Object],[object Object],[object Object],[object Object]" depth="0"></drop-down-items-stub>
      </li>
      <li class="nav-item dropdown"><a id="plugins" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" class="nav-link dropdown-toggle">
          Plugins
        </a>
        <drop-down-items-stub parent="[object Object]" items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]" depth="0"></drop-down-items-stub>
      </li>
      <li class="nav-item dropdown"><a id="admin" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" class="nav-link dropdown-toggle">
          Admin
        </a>
        <drop-down-items-stub parent="[object Object]" items="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]" depth="0"></drop-down-items-stub>
      </li>
      <li class="nav-item"><a href="/menu/main/useraccount" class="nav-link">
          Account
        </a></li>
    </ul>
    <ul class="navbar-nav justify-content">
      <!---->
    </ul>
    <ul class="navbar-nav justify-content-end">
      <li class="nav-item"><a href="https://molgenis.gitbooks.io/molgenis/content/" target="_blank" class="nav-link">
          Help
        </a></li>
      <li class="nav-item">
        <form id="logout-form" method="post" action="/logout" class="form-inline"><button id="signout-button" type="button" class="btn btn-outline-secondary">
            Sign out
          </button></form>
      </li>
    </ul>
  </div>
</nav>
`;
